### PR TITLE
NO-ISSUE Remove kube job references and combine job types

### PR DESCRIPTION
**Pull request by carbonin**
This unused code has been sitting around for 6 months or so (the actual use of a kubernetes job was removed in https://github.com/openshift/assisted-service/commit/e805401970e98676bd145343952dea949ca5b4a3)

The only real difference between the local and regular job
implementations is an assumption that we're using the local s3 client.
Before this change that was incredibly confusing because in the
implementation of the jobs used the s3 client as an interface rather
than a particular implementation.

This leans into the interface model and removes the reliance on
the generator working directory needing to be the same as the local s3
client base directly. This is done by copying the files after they are
generated using generator.UploadToS3.

Hopefully this makes this whole area of the code base a bit less confusing

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -531,23 +524,6 @@ func uploadBootFiles(objectHandler s3wrapper.API, openshiftVersionsMap models.Op
 	return errs.Wait()
 }
 
-func newISOInstallConfigGenerator(log *logrus.Entry, objectHandler s3wrapper.API, operatorsApi operators.API) generator.ISOInstallConfigGenerator {
-	var configGenerator generator.ISOInstallConfigGenerator
-	switch Options.Storage {
-	case storage_s3:
-		kclient, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme.Scheme})
```
Comment (djzager) [inline cmd/main.go:538]: Is it alright that I don't see `kclient` anywhere in the new `generator` code?

**Review by djzager — COMMENTED** [review]
Visually this looks good to me. When it's out of WIP I'll want to have another look.

File: cmd/main.go (hunk 1 above)
Comment (carbonin) [reply cmd/main.go:538]: Yeah, that was part of the goal. We don't talk to the cluster during this process anymore.

**Review by filanov — APPROVED** [review]

**Review by djzager — APPROVED** [review]

**Pull request merged**